contracts/l1/Bridge.sol

```diff
@@ -442,4 +466,79 @@ contract Bridge is IBridge, VersionedInitializable {
         }
         revert(Errors.B_NOT_ENOUGH_REWARDS);
     }
+
+    function startDepositCancellation(
```

Comment:
users can get all the needed data from the emitted event on the starknetMessaging contract [here](https://github.com/starkware-libs/starkgate-contracts/blob/c08863a1f08226c09f1d0748124192e848d73db9/src/starkware/starknet/solidity/StarknetMessaging.sol#L107)

Reply:
Yes, but what if the deposit is a smart contract? It is not so good idea to depend only on events.
Also, from what I see on the StarkNetMessaging side, you can use `msgHash` to get the message data (not sure if the nonce is needed). Why not getting the message hash, return it on the deposit()?

Reply:
The function [cancelL1toL2Message](https://github.com/starkware-libs/starkgate-contracts/blob/c08863a1f08226c09f1d0748124192e848d73db9/src/starkware/starknet/solidity/StarknetMessaging.sol#L147) takes the payload+selector+nonce as params and then computes the hash, so even if we return the msgHash to the user it won't be useful since he can't get the payload nor the other hashed data from it. Hence, the need to explicitly provide the payload on `cancelDeposit`, what we can do instead as a security measure is return the nonce only on deposit because it's critical to restoring the L2 message.

Reply:
I think at least there should be a clearly defined set of steps (on smart contracts or not) on how to get the parameters to start a cancellation.
Because multiple come from different sources (blocknumber, variables on contract, inputs), so it can get quite complicated